core/src/test/java/org/mskcc/oncotree/utils/OncotreeTestConfig.java

```diff
@@ -202,20 +214,51 @@ private Version setupMockVersion() {
         List<OncoTreeNode> tmpOncoTreeRepositoryMockResponse = new ArrayList<>();
         for (int pos = 0; pos < rawTestValueSource.length; pos = pos + valuesPerCase) {
             OncoTreeNode nextNode = new OncoTreeNode();
-            nextNode.setURI("TESTURI");
             nextNode.setCode(rawTestValueSource[pos]);
             nextNode.setName(rawTestValueSource[pos + 1]);
             nextNode.setMainType(rawTestValueSource[pos + 2]);
             nextNode.setColor(rawTestValueSource[pos + 3]);
             nextNode.setParentCode(rawTestValueSource[pos + 4]);
+            nextNode.setURI(rawTestValueSource[pos + 5]);
             tmpOncoTreeRepositoryMockResponse.add(nextNode);
         }
         return tmpOncoTreeRepositoryMockResponse;
     }
 
+    private List<OncoTreeNode> setupOncotreeRepositoryMockResponseHistoryChange() {
```

Comment:
we may want to create a general function for applying sets of modifications at particular timepoints. The destination version could be an argument to this function as well as the prior set of oncotree nodes, and only the changes intended to happen at that point in time would be applied to construct the new version.